tests/test_core_network.py

```diff
@@ -500,12 +516,36 @@ def test_simplifing_puma_network_results_in_correct_record_of_removed_links_and_
     assert report['schedule']['schedule_level']['is_valid_schedule']
 
 
-def test_simplified_network_saves_to_correct_dtds(tmpdir, network_dtd, schedule_dtd):
-    n = read.read_matsim(path_to_network=puma_network_test_file, epsg='epsg:27700',
-                         path_to_schedule=puma_schedule_test_file)
+def test_simplify_does_not_oversimplify_PT_endpoints(
+        puma_network_pre_simplify, puma_network_post_simplify):
+    n = puma_network_pre_simplify
+
+    stops_at_risk = ['5221390681543854913', '5221390302070799085', '5221390323679791901']
```

Comment:
Are these PT stop IDs? Maybe they should belong to the fixture, so you have, say, a `puma_network_with_pt_stops` fixture that is the network and a list of the PT stop IDs that belong to the network. That way the implicit knowledge of the network, i.e. what the IDs of the PT stop nodes are, is contained in just one place - the fixture - rather than leaking out into the test (or more than one test).

So the test would look more like (with a helper method `assert_link_length` - maybe there is a better, higher level version of that, like `assert_is_blah`, that hides the detail of `link length == 1` -  `assert_valid_pt_node`, or whatever):

```python
def test_simplify_does_not_oversimplify_PT_endpoints(puma_network_with_pt_stops):
    network, pt_stop_ids = puma_network_with_pt_stops
    for stop in pt_stop_ids:
        assert_link_length(stop, 1, network)
        
    network.simplify()
    
     for stop in pt_stop_ids:
         assert_link_length(stop, 1, network)
```

That looks more like the usual basic structure of a unit test:

- setup
- call code under test
- validate expectations

Also known as "Arrange, Act, Assert"

Or... maybe there is a way you can query the network and discover the PT stops, inside the test, rather than having a pre-prepared list of the PT stop IDs? So then, with a helper method `get_pt_stops`,  something like:

```python
def test_simplify_does_not_oversimplify_PT_endpoints(puma_network_with_pt_stops):
    for stop_id in get_pt_stops(puma_network_with_pt_stops):
        assert_link_length(stop_id, 1, puma_network_with_pt_stops)
        
    puma_network_with_pt_stops.simplify()
    
    for stop_id in get_pt_stops(puma_network_with_pt_stops):
        assert_link_length(stop_id, 1, puma_network_with_pt_stops)
```

Reply:
nah, so these are only a small subset of specific stops that were affected by the bug, the majority of other PT stops were fine. I guess there is a way to pull out all of those stops with some conditions, but that will require using non trivial amount of genet code which we're not testing here, so I'm not sure which is better ?

Reply:
moved to a fixture with a descriptive name